core/deployment/src/main/java/io/quarkus/deployment/pkg/PackageConfig.java

```diff
@@ -11,25 +11,75 @@
 @ConfigRoot
 public class PackageConfig {
 
+    /**
+     * @deprecated Use {@link Type#JAR} instead
+     */
+    @Deprecated
     public static final String JAR = "jar";
+    /**
+     * @deprecated Use {@link Type#UBER_JAR} instead
+     */
+    @Deprecated
     public static final String UBER_JAR = "uber-jar";
+    /**
+     * @deprecated Use {@link Type#FAST_JAR} instead
+     */
+    @Deprecated
     public static final String FAST_JAR = "fast-jar";
+    /**
+     * @deprecated Use {@link Type#MUTABLE_JAR} instead
+     */
+    @Deprecated
     public static final String MUTABLE_JAR = "mutable-jar";
     /**
      * @deprecated use 'legacy-jar' instead
      */
     @Deprecated
     public static final String LEGACY = "legacy";
+    /**
+     * @deprecated Use {@link Type#LEGACY_JAR} instead
+     */
+    @Deprecated
     public static final String LEGACY_JAR = "legacy-jar";
+    /**
+     * @deprecated Use {@link Type#NATIVE} instead
+     */
+    @Deprecated
     public static final String NATIVE = "native";
+    /**
+     * @deprecated Use {@link Type#NATIVE_SOURCES} instead
+     */
+    @Deprecated
     // does everything 'native' but stops short of actually executing the 'native-image' command
     public static final String NATIVE_SOURCES = "native-sources";
 
+    public enum Type {
```

Comment:
```suggestion
    public enum BuiltInType {
```

Reply:
Or perhaps `PackagingType`?

Reply:
Probably then as an enum outside of the `PackageConfig` class?

Reply:
Going with `BuildInType` for now, it reads nicely as `PackageConfig.BuiltInType` if being referenced with the containing class name.